From 612b6846de986c98f906d2b42857c869cd06a606 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 16 Mar 2020 05:53:28 +0200
Subject: [PATCH] [SW-0] goya2: turn preboot_sync_btl weak

Seem like in onix there is an issue with boot manager which is not going
back to idle state so we hang for good waiting for this to happen

Change-Id: Ic6dbd962053531db1ffe18df490f15ebe10266e0
---

diff --git a/goya2/common.c b/goya2/common.c
index c48a56a..233c339 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -190,7 +190,7 @@
  * Wait for Boot Manager to complete PCIe image execution
  * and return to idle state, so it won't initiate UART access anymore
  */
-void preboot_sync_btl(void)
+__attribute__((weak)) void preboot_sync_btl(void)
 {
 	while (regmap_readl(mmPSOC_GLOBAL_CONF_BTM_FSM))
 		;
diff --git a/goya2/onix.c b/goya2/onix.c
index c7abdd0..026db64 100644
--- a/goya2/onix.c
+++ b/goya2/onix.c
@@ -37,3 +37,5 @@
 	/* baud rate doesn't support odd number */
 	return (factor + 1) & 0xFFFE;
 }
+
+void preboot_sync_btl(void) {}
